plugin/src/main/java/org/opensearch/ml/sdkclient/SdkClientModule.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.ml.sdkclient;
+
+import org.opensearch.OpenSearchException;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.transport.aws.AwsSdk2Transport;
+import org.opensearch.client.transport.aws.AwsSdk2TransportOptions;
+import org.opensearch.common.inject.AbstractModule;
+import org.opensearch.core.common.Strings;
+import org.opensearch.sdk.SdkClient;
+
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.regions.Region;
+
+/**
+ * A module for binding this plugin's desired implementation of {@link SdkClient}.
+ */
+public class SdkClientModule extends AbstractModule {
+
+    public static final String REMOTE_METADATA_ENDPOINT = "REMOTE_METADATA_ENDPOINT";
+    public static final String REGION = "REGION";
+
+    private final String remoteMetadataEndpoint;
+    private final String region;
+
+    /**
+     * Instantiate this module using environment variables
+     */
+    public SdkClientModule() {
+        this(System.getenv(REMOTE_METADATA_ENDPOINT), System.getenv(REGION));
+    }
+
+    /**
+     * Instantiate this module specifying the endpoint and region. Package private for testing.
+     * @param remoteMetadataEndpoint The remote endpoint
+     * @param region The region
+     */
+    SdkClientModule(String remoteMetadataEndpoint, String region) {
+        this.remoteMetadataEndpoint = remoteMetadataEndpoint;
+        this.region = region;
+    }
+
+    @Override
+    protected void configure() {
+        boolean local = Strings.isNullOrEmpty(remoteMetadataEndpoint);
```

Comment:
Looks good for initial implementation. Lets add a follow up task to support other types of implementations. Ideally adding a new implementation should not require changes to plugin code base.

Reply:
Adding a new implementation requires adding the implementation class always, so I don't think there's any way to get around updating this class to choose from among the options based on whatever criteria we decide on (settings, environment variable, command line argument, etc.).  Not sure how we can future-proof this as it's an extension point that runs at startup.

Reply:
Adding new implementations to plugin repo could increase the size of the plugin. Instead we should be able to dynamically load implementation during runtime. Ideally these implementations should be loaded during runtime.

Reply:
> Instead we should be able to dynamically load implementation during runtime. 

Agreed, particularly since there's a lot of specific configuration around these (user credentials, access keys, SSL keystore and keystore password, etc.). 

For now I've just switched this to a basic http RestClient implementation to get the rest of the code working, and will circle back to properly doing this later.